lib/spack/spack/platforms/cray.py

```diff
@@ -2,254 +2,10 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-import os
 import os.path
-import platform
-import re
-
-import archspec.cpu
-
-import llnl.util.tty as tty
-from llnl.util.symlink import readlink
-
-import spack.target
-import spack.version
-from spack.operating_systems.cray_backend import CrayBackend
-from spack.operating_systems.cray_frontend import CrayFrontend
-from spack.paths import build_env_path
-from spack.util.executable import Executable
-from spack.util.module_cmd import module
-
-from ._platform import NoPlatformError, Platform
-
-_craype_name_to_target_name = {
-    "x86-cascadelake": "cascadelake",
-    "x86-naples": "zen",
-    "x86-rome": "zen2",
-    "x86-milan": "zen3",
-    "x86-skylake": "skylake_avx512",
-    "mic-knl": "mic_knl",
-    "interlagos": "bulldozer",
-    "abudhabi": "piledriver",
-}
-
-_ex_craype_dir = "/opt/cray/pe/cpe"
-_xc_craype_dir = "/opt/cray/pe/cdt"
 
 
 def slingshot_network():
```

Comment:
I kind of agree. I left this here because the function is used in a few packages, so if we move it we might break custom repos. If it's fine to do so, I totally agree we should move it. 

Another alternative is to leave a wrapper here with a warning saying the function moved elsewhere.